test/src/test/java/org/corfudb/integration/CorfuReplicationUpgradeIT.java

```diff
@@ -937,4 +939,46 @@ private void verifyVersion(CorfuStore corfuStore, String expectedVersion, boolea
         Assert.assertEquals(isUpgraded, actualUpgradedFlag);
         log.info("Verified version");
     }
+
+    /**
+     * Code coverage test for the simple LRRollingUpgradeHandler to test if we are able to successfully
+     * 1. Simulate startRollingUpgrade()
+     * 2. verify that rolling upgrade is detected
+     * 3. simulate endRollingUpgrade()
+     * 4. verify that data migration and rolling upgrade end is detected
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testLocalClusterRollingUpgrade() throws Exception {
+        log.info(">> Setup replication for testing during rolling upgrade of active cluster");
+        setupActiveAndStandbyCorfu();
+
+        DefaultAdapterForUpgradeActive.startRollingUpgrade(corfuStoreActive);
```

Comment:
If we instantiate the default adapter before starting rolling upgrade, do we need to make start and endRollingUpgrade static?

Reply:
good point - no they need not be - let me make them non-static. thanks